### PR TITLE
negative test for changing domain name using MII dynamic update 

### PR DESCRIPTION
**Pull request by xiancao**
This is a negative test for MII dynamic update - changing domain name at runtime (unsupported).

Jenkins result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3337/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3340/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3343/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 1]
```diff
@@ -453,6 +516,65 @@ private void verifyIntrospectorRuns() {
...
+  private String getPodNameFromJobName(String namespace, String jobName) {
+    String labelSelector = String.format("weblogic.domainUID in (%s)", domainUid);
+    V1Pod introspectorPod = assertDoesNotThrow(() -> getPod(namespace, labelSelector, jobName));
+    assertNotNull(introspectorPod);
+    assertNotNull(introspectorPod.getMetadata());
+    return introspectorPod.getMetadata().getName();
+  }
+
+  private boolean podLogContainsExpectedErrorMsg(String introspectJobName, String namespace, String errormsg) {
+
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:563]: > 
```suggestion
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 2]
```diff
@@ -453,6 +516,65 @@ private void verifyIntrospectorRuns() {
...
+    String introspectPodName = getPodNameFromJobName(namespace, introspectJobName);
+
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:565]:
```suggestion
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 3]
```diff
@@ -453,6 +516,65 @@ private void verifyIntrospectorRuns() {
...
+                        + "Elapsed time {0}ms, remaining time {1}ms",
+                    condition.getElapsedTimeInMS(), condition.getRemainingTimeInMS()))
+        .until(() ->
+            podStatusPhaseContainsString(domainNamespace, introspectJobName, "Failed"));
+  }
+
+  private String getPodNameFromJobName(String namespace, String jobName) {
+    String labelSelector = String.format("weblogic.domainUID in (%s)", domainUid);
+    V1Pod introspectorPod = assertDoesNotThrow(() -> getPod(namespace, labelSelector, jobName));
+    assertNotNull(introspectorPod);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:557]: add message

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 4]
```diff
@@ -453,6 +516,65 @@ private void verifyIntrospectorRuns() {
...
+    assertNotNull(introspectorPod.getMetadata());
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:558]: add message

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 5]
```diff
@@ -443,7 +452,61 @@ public void testMiiRemoveTarget() {
...
+  }
+
+  private boolean domainStatusConditionMsgContainsErrorMsg(Domain domain, String errorMsg) {
+    for (DomainCondition domainCondition : domain.getStatus().getConditions()) {
+      if (domainCondition.getType().equalsIgnoreCase("Failed")
+          && domainCondition.getMessage().contains(errorMsg)) {
+        return true;
+      }
+    }
+
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:507]:
```suggestion
```

**Review by vanajamukkara — APPROVED** [review]
LG

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 6]
```diff
@@ -443,7 +452,61 @@ public void testMiiRemoveTarget() {
...
+    // Replace contents of an existing configMap
+    replaceConfigMapWithModelFiles(configMapName, domainUid, domainNamespace,
+        Arrays.asList(pathToChangeDomainNameYaml.toString()), withStandardRetryPolicy);
+
+    // Patch a running domain with introspectVersion.
+    patchDomainResourceWithNewIntrospectVersion(domainUid, domainNamespace);
+
+    // Verifying introspector pod is created and failed
+    logger.info("verifying the introspector failed and the pod log contains the expected error msg");
+    String expectedErrorMsg = "name is not one of the attribute names allowed in model location topology";
```
Comment (jshum2479) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:483]: I don't think this is the correct message.  Are you wait for the latest WDT release 1.9.8 (not yet released but should be today).

Comment (xiancao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:483]: I tried with latest WDT from https://objectstorage.us-phoenix-1.oraclecloud.com/n/weblogick8s/b/wko-system-test-files/o/weblogic-deploy-master.zip. I got the same error msg. What's the new error msg I should get?

**Pull request merged**